src/Components/src/Microsoft.AspNetCore.Components/ParameterCollection.cs

```diff
@@ -196,5 +197,25 @@ internal void CaptureSnapshot(ArrayBuilder<RenderTreeFrame> builder)
                 builder.Append(_frames, _ownerIndex + 1, numEntries);
             }
         }
+
+        /// <summary>
+        /// Creates a new <see cref="ParameterCollection"/> from the given <see cref="IDictionary{TKey, TValue}"/>.
+        /// </summary>
+        /// <param name="parameters">The <see cref="IDictionary{TKey, TValue}"/> with the parameters.</param>
+        /// <returns>A <see cref="ParameterCollection"/>.</returns>
+        public static ParameterCollection FromDictionary(Dictionary<string, object> parameters)
```

Comment:
Use `IDictionary` as the comment suggests?
Throw `ArgumentNullException` if `parameters` is `null`?